### PR TITLE
Fix #180562 totaljobs.com

### LINKED ISSUE #180562: www.totaljobs.com
### Issue URL (Annoyance)
 [https://www.totaljobs.com/](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Fwww.totaljobs.com%2F)

### Screenshots
<details>
  <summary>Screenshot 1 </summary>

  Screenshot 1
</details>

### System configuration

Information | Value
--- | ---
AdGuard product: | AdGuard Browser Extension v4.3.46
System version: | Windows 10
Browser: | Firefox
Stealth mode: | disabled
Filters: | <b>Ad Blocking:</b><br/>AdGuard Base<br/><br/><b>Privacy:</b><br/>AdGuard Tracking Protection<br/><br/><b>Social Widgets:</b><br/>AdGuard Social Media<br/><br/><b>Annoyances:</b><br/>AdGuard Cookie Notices, <br/>AdGuard Popups, <br/>AdGuard Other Annoyances<br/><br/><b>Language-specific:</b><br/>AdGuard Russian

### PR DESCRIPTION
**Pull request by grumaxxx**
Fix #180562

### CODE REVIEW

**Review by piquark6046 — CHANGES_REQUESTED** [review]
App banner: `article[data-testid="hp-app-banner-holder"]`
Possible analytics: `https://www.totaljobs.com/analytics/api/v1/tracking`, `collect.stepstone.co.uk` and `/^https:\/\/www\.totaljobs\.com\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+/`

**Comment by AdamWr** [comment]
Regarding the last rule, I mean this:
```adblock
/^https:\/\/www\.totaljobs\.com\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+/
```
Maybe it could be specified to `script` only and with `domain` modifier?
Like:
```adblock
/^https:\/\/www\.totaljobs\.com\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+/$script,domain=totaljobs.com
```

**Comment by piquark6046** [comment]
https://github.com/AdguardTeam/AdguardFilters/pull/180806#issuecomment-2149212094

It seems that `$xmlhttprequest` modifier also needs to be added.

**Comment by grumaxxx** [comment]
> App banner: `article[data-testid="hp-app-banner-holder"]` Possible analytics: `https://www.totaljobs.com/analytics/api/v1/tracking`, `collect.stepstone.co.uk` and `/^https:\/\/www\.totaljobs\.com\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+/`

Couldn't find the relevant information, is there a special place for rules related to analytics?

**Comment by AdamWr** [comment]
I think that this one:
```adblock
||collect.stepstone.co.uk^
```
can be added to - https://github.com/AdguardTeam/AdguardFilters/blob/master/SpywareFilter/sections/tracking_servers_firstparty.txt
And these:
```adblock
||totaljobs.com/analytics/api/v1/tracking
/^https:\/\/www\.totaljobs\.com\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+\/[a-zA-Z0-9]+/$script,xmlhttprequest,domain=totaljobs.com
```
to - https://github.com/AdguardTeam/AdguardFilters/blob/master/SpywareFilter/sections/specific.txt

**Comment by piquark6046** [comment]
https://github.com/AdguardTeam/AdguardFilters/pull/180806#issuecomment-2149235508
Please refer the following screencast:

<details><summary>Screencast</summary>
<p>

[Screencast from 2024-06-05 17-49-57.webm](https://github.com/AdguardTeam/AdguardFilters/assets/98787049/488d53b2-097b-4003-abc5-3ebc3d9d691b)

</p>
</details>

**Comment by AdamWr** [comment]
As far as I can see, this request is created by a blocked script, so blocking just `script` should be enough, but adding `xmlhttprequest` to this rule is fine.

**Review by piquark6046 — CHANGES_REQUESTED** [review]
Can you relocate the new rules to top of each section?

File: BaseFilter/sections/specific.txt [hunk 1]
```diff
@@ -1749,6 +1749,7 @@ champion.gg##.aside-content-column
 lolchess.gg##div[class^="ap_"].profile
 lolchess.gg##div[class*="Ad"]
 lolchess.gg##.ap_top
+totaljobs.com##article[data-testid="hp-app-banner-holder"]
```
Comment (AdamWr) [inline BaseFilter/sections/specific.txt:1752]: Please move this rule to - https://github.com/AdguardTeam/AdguardFilters/blob/master/AnnoyancesFilter/MobileApp/sections/mobile-app_specific.txt

**Review by AdamWr — CHANGES_REQUESTED** [review]

**Review by piquark6046 — APPROVED** [review]

**Review by AdamWr — APPROVED** [review]

**Pull request merged**